pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4301,10 +4301,11 @@ public static CompletableFuture<PartitionedTopicMetadata> getPartitionedTopicMet
                                 }).exceptionally(ex -> {
                                     Throwable throwable2 = FutureUtil.unwrapCompletionException(ex);
                                     if (throwable2 instanceof RestException) {
-                                        log.warn("Failed to authorize {} on topic {}", clientAppId, topicName);
-                                        authorizationFuture.completeExceptionally(new PulsarClientException(
-                                                String.format("Authorization failed %s on topic %s with error %s",
-                                                clientAppId, topicName, throwable2.getMessage())));
```

Comment:
This is in the broker.
I thinks that we should split the path per component

Reply:
> This is in the broker. I thinks that we should split the path per component

Sorry, could you please tell more specific? I'm guessing you are telling we shouldn't use exception class (`PulsarClientException` in here) that defined in the another component? Am I right?

Reply:
@eolivelli Could you explain in details? I'm also wondering what did you mean by `split the path per component`.